src/dependency-container.ts

```diff
@@ -111,13 +156,19 @@ export class DependencyContainer implements Types.DependencyContainer {
       if (isValueProvider(registration.provider)) {
         return registration.provider.useValue;
       } else if (isTokenProvider(registration.provider)) {
-        return registration.options.singleton ?
-          (registration.instance || (registration.instance = this.resolve(registration.provider.useToken))) :
-          this.resolve(registration.provider.useToken);
+        return registration.options.singleton
+          ? registration.instance ||
```

Comment:
I always found putting the `?` at the start of the line odd - From a syntax perspective, the "question" applies to the conditional, so should be on the same line as it.

Reply:
Oh, I see - this is prettier's opinion. Fair enough.

Reply:
when we add a `.prettierrc` I think this will get better